### PR TITLE
Adding hint on how to fix the 'not instrumented' error message

### PR DESCRIPTION
**Pull request by mkit**
Providing a hint to a developer on test failure caused by missing Quasar java agent.

### CODE REVIEW

**Comment by mikehearn** [comment]
I am not convinced this is correct. Have you tested it? 

The issue reported to us arises when the magic command line argument is missing from the JVM startup command. This can't happen with the driver because the driver code controls the JVM command line. It only really happens when people run unit tests from inside IntelliJ. So the right place to catch this exception is in the state machine manager, as that's the one place that's guaranteed to be on the stack at all the places where this error might occur.

**Comment by mkit** [comment]
Yes, I tested it by running my integration test (from the IntelliJ), without the -javaagent argument set in the run configuration. I might be confusing things so please correct me. See the screenshot of the IntelliJ console output, when running the test.

screen shot 2017-08-17 at 16 50 04

**Comment by mikehearn** [comment]
OK, so that test is using the driver in in-process mode, so I can see why it doesn't control the command line arguments there.

The original bug report was about IOUFlowTests (in cordapp-tutorial), which uses MockNetwork. In that case there's no driver. That's why I suggest putting the check closer to the original source of the exception, in the StateMachineManager. Otherwise the error won't get printed successfully when a MockNetwork based test has the same issue.

**Comment by exFalso** [comment]
To clarify, this error can occur in any test that uses flows in a JVM without the javaagent. This includes in-process drivers as well as mocknetwork-based tests.
By default intellij doesn't include the javaagent flag in the individual test run configurations, so this error will occur to anyone running an individual test. This reminds me, we could add the flag to the default JUnit run configuration.

We could put a `RuntimeMXBeanWhatever.vmOptions` check somewhere like `StateMachineManager.add`

**Comment by mkit** [comment]
I have put a check into the StateMachineManager (thanks @mikehearn) based on the quasar one. Please check.

File: node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt [hunk 1]
```diff
@@ -166,11 +167,21 @@ class StateMachineManager(val serviceHub: ServiceHubInternal,
...
         listenToLedgerTransactions()
         serviceHub.networkMapCache.mapServiceRegistered.then { executor.execute(this::resumeRestoredFibers) }
     }
 
+    private fun checkQuasarJavaAgentPresence() {
+        if (!SuspendableHelper.isJavaAgentActive()) {
+            val message = """Missing the '-javaagent' JVM argument.
+                #Make sure you run the tests with the Quasar java agent attached to your JVM.
+                #See https://docs.corda.net/troubleshooting.html - 'Fiber classes not instrumented' for more details."""
+            throw IllegalArgumentException(message)
```
Comment (mikehearn) [inline node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt]: throw AssertionError, not IllegalArgumentException.

Comment (mkit) [inline node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt]: Good spot. Do you think IllegalStateException would be better? I want to use the `check` method from the kotlin's stdlib.

Comment (mikehearn) [inline node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt]: Sure, that works too.

**Pull request merged**